js/ui/number_box/number_box.mask.js

```diff
@@ -234,7 +234,18 @@ var NumberBoxMask = NumberBoxBase.inherit({
         return edited;
     },
 
+    _isNumberVeryLong: function(text) {
+        var decimalSeparator = number.getDecimalSeparator(),
+            regExp = new RegExp("[^0-9" + decimalSeparator + "]", "g");
+
+        return text.replace(regExp, "").length > MAXIMUM_FLOAT_LENGTH;
+    },
+
     _parseNumber: function(text) {
+        if(this._isNumberVeryLong(text)) {
```

Comment:
By my opinion, this code can be refactored as in the following example:
```
if(!this._isNumberVeryLong(text)) {
	var format = this._getFormatPattern();
    return number.parse(text, format);
}
```

Reply:
It may lead to bugs because it should be visually that method returns undefined

Reply:
What bugs can there be?